aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1137,41 +1141,38 @@ Tensor _lu_solve_helper_cpu(const Tensor& self, const Tensor& LU_data, const Ten
   return self_working_copy;
 }
 
+// Supports arbitrary batch dimensions for self and LU_data (implicity LU_pivots also)
 Tensor lu_solve(const Tensor& self, const Tensor& LU_data, const Tensor& LU_pivots) {
-  TORCH_CHECK(self.dim() == 3 || self.dim() == 2,
-              "b should have 2 or 3 dimensions, but has ", self.dim(), " dimensions instead");
-  TORCH_CHECK(LU_data.dim() == 3,
-              "LU_data should have 3 dimensions, but has ", LU_data.dim(), " dimensions instead");
-  TORCH_CHECK(self.size(0) == LU_data.size(0),
-              "b and LU_data should have the same number of batches");
-  TORCH_CHECK(LU_pivots.size(1) == LU_data.size(2),
+  TORCH_CHECK(self.dim() >= 2,
+              "b should have at least 2 dimensions, but has ", self.dim(), " dimensions instead");
```

Comment:
Might be good to give the size as well. The more information, the easier users can identify their tensors

Reply:
Sure.

Reply:
I'll have to address this for other methods, so is it fine if I do this in a follow-up PR?

Reply:
Yes, that would be great.